File: 12-integer-to-roman/12-integer-to-roman.java
class Solution {
    private static final int[] values = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};    
    private static final String[] symbols = {"M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"};

    public String intToRoman(int num) {
        StringBuilder res = new StringBuilder();
        
        for(int i=0; i<13; i++)
        {
            while(num >= values[i])
            {
                num -= values[i];
                res.append(symbols[i]);
            }
        }
        
        return res.toString();  
    }
}

File: 12-integer-to-roman/12-integer-to-roman.cpp
class Solution {
public:
    string intToRoman(int num) {
        string romans[] = {"I","IV","V","IX","X","XL","L","XC","C","CD","D","CM","M"};
        int numerals[]={1,4,5,9,10,40,50,90,100,400,500,900,1000};
        
        string s;
        while(num)
        {
            for(int i = 12; i >= 0; i--)
            {
                int x = num/numerals[i];
                if(x)
                {
                    while(x--)
                    {
                        s += romans[i];
                    }
                    num %= numerals[i];
                    break;
                }
            }
        }
        return s;
    }
};